# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../tetex/getline.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Avoid conflicts with getline function from glibc

--- ./texk/dvipsk/afm2tfm.c.orig	2010-07-26 16:10:16.828270364 +0200
+++ ./texk/dvipsk/afm2tfm.c	2010-07-26 16:11:03.413007135 +0200
@@ -257,7 +257,7 @@
 }
 
 int
-getline P1H(void) {
+afm2tfm_getline P1H(void) {
    register char *p ;
    register int c ;
 
@@ -606,7 +606,7 @@
    ai = newchar() ;
    ai->adobenum = -1 ;
    ai->adobename = "||" ; /* boundary character name */
-   while (getline()) {
+   while (afm2tfm_getline()) {
       switch(interest(paramstring())) {
 case FontName:
          fontname = paramnewstring() ;
@@ -1882,7 +1882,7 @@
 
    while (1) {
       while (param == 0 || *param == 0) {
-         if (getline() == 0)
+         if (afm2tfm_getline() == 0)
             error("! premature end in encoding file") ;
          for (p=buffer; *p; p++)
             if (*p == '%') {
@@ -1973,7 +1973,7 @@
       p = gettoken() ;
       if (strcmp(p, "]"))
          error("! token 258 in encoding must be make-array (])") ;
-      while (getline()) {
+      while (afm2tfm_getline()) {
          for (p=buffer; *p; p++)
             if (*p == '%') {
                if (ignoreligkern == 0)
--- ./texk/web2c/cpascal.h.orig	2010-07-26 16:17:28.940755570 +0200
+++ ./texk/web2c/cpascal.h	2010-07-26 16:18:30.988518578 +0200
@@ -241,6 +241,12 @@
 #define getname vms_getname
 #endif
 
+
+#ifdef getline
+#undef getline
+#endif
+#define getline web2c_getline
+
 /* Declarations for the routines we provide ourselves in lib/.  */
 
 extern string basenamechangesuffix P3H(const_string,const_string,const_string);
